application/CohortManager/src/Functions/CohortDistributionServices/RetrieveCohortDistribution/RetrieveCohortDistribution.cs

```
@@ -44,4 +51,26 @@ public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Ano
             return _createResponse.CreateHttpResponse(HttpStatusCode.InternalServerError, req);
         }
     }
+
+    private static int GetQueryParameterAsInt(HttpRequestData req, string key, int defaultValue = 0)
+    {
+        var queryString = req.Query[key];
+        return int.TryParse(queryString, out int value) ? value : defaultValue;
+    }
+
+    private static int GetRowCount(HttpRequestData req)
+    {
+        return GetQueryParameterAsInt(req, "rowCount");
+    }
+
+    private static int GetServiceProviderId(HttpRequestData req)
+    {
+        return GetQueryParameterAsInt(req, "serviceProviderId");
+    }
+
+    private HttpResponseData LogErrorResponse(HttpRequestData req, string errorMessage)
+    {
+        _logger.LogError(errorMessage);
+        return _createResponse.CreateHttpResponse(HttpStatusCode.BadRequest, req);
```

I think we should also return the error message as the body of the response?

---

error message added along with the status code and request